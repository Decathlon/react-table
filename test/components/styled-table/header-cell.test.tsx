/// <reference path="../../typings/tests-entry.d.ts" />

import * as React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import HeaderCell from "../../../src/components/styled-table/header-cell";

describe("HeaderCell component", () => {
  test("should render the default HeaderCell", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<HeaderCell title="Foo" value="Bar" badge="30" />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render the default HeaderCell (with subtitle)", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<HeaderCell title="Foo" value="Bar" badge="30" />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render the HeaderCell (with custom className)", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<HeaderCell title="Foo" value="Bar" badge="30" className="custom" />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render the current HeaderCell", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<HeaderCell title="Foo" value="Bar" badge="30" isCurrent />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
});
