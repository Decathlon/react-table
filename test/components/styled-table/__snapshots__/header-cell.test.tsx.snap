// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderCell component should render the current HeaderCell 1`] = `
<div
  className="header-cell header-cell-current"
>
  <React.Fragment>
    <div
      className="header-cell-line"
    />
    <UNDEFINED
      badge="30"
      className="header-cell-bubble"
    >
      <div
        className="header-cell-data"
      >
        <div
          className="header-cell-title"
        >
          Foo
        </div>
        <div
          className="header-cell-value"
        >
          Bar
        </div>
      </div>
    </UNDEFINED>
  </React.Fragment>
</div>
`;

exports[`HeaderCell component should render the default HeaderCell (with subtitle) 1`] = `
<div
  className="header-cell"
>
  <div
    className="header-cell-data"
  >
    <div
      className="header-cell-title"
    >
      Foo
    </div>
    <div
      className="header-cell-value"
    >
      Bar
    </div>
  </div>
</div>
`;

exports[`HeaderCell component should render the default HeaderCell 1`] = `
<div
  className="header-cell"
>
  <div
    className="header-cell-data"
  >
    <div
      className="header-cell-title"
    >
      Foo
    </div>
    <div
      className="header-cell-value"
    >
      Bar
    </div>
  </div>
</div>
`;
